sdk/file_sdk/Common/file_format_factory.cpp

```diff
@@ -0,0 +1,81 @@
+#include "file_format_factory.h"
+#include <algorithm>
+#include <vector>
+#include <RMSExceptions.h>
+#include "file_format.h"
+#include "XMPFileFormat/xmp_file_format.h"
+#include "CompoundFileFormat/compound_file_format.h"
+#include "OPCFileFormat/opc_file_format.h"
+#include "PDFFileFormat/pdf_file_format.h"
+#include "PFileFileFormat/pfile_file_format.h"
+#include "DefaultFormat/default_format.h"
+
+using std::vector;
+using mip::file::FileFormat;
+using mip::file::CompoundFileFormat;
+using mip::file::DefaultFormat;
+using mip::file::OPCFileFormat;
+using mip::file::PDFFileFormat;
+using mip::file::PFileFileFormat;
+using mip::file::XMPFileFormat;
+
+namespace {
+static const vector<string> kXMPExtentions {".jpg", ".jpeg", ".jpe", ".jif", ".jfif", ".jfi",".png", ".tif", ".tiff", ".gif", ".dng", ".psd"};
+static const vector<string> kCompoundFileExtentions {".doc", ".doc", ".xls", "xlt", ".ppt", ".pps", ".pot", ".vsd", ".vdw", ".vst", ".pub", ".sldprt", ".slddrw", ".sldasm", ".mpp", ".mpt"};
+static const vector<string> kOPCFileExtentions {".docx", ".docm", ".dotm", ".dotx", ".xlsx", ".xltx", ".xltm", ".xlsm", ".xlsb", ".pptx", ".ppsx", ".pptm", ".ppsm", ".potx", ".potm",  ".vsdx", ".vsdm", ".vssx", ".vssm", ".vstx", ".vstm", ".xps", ".oxps", ".dwfx"};
+static const vector<string> kPDFExtentions {".pdf"};
+static const vector<string> kPfileExtentions {".pfile", ".ppdf", ".ptxt", ".pxml", ".pjpg", ".pjpeg", ".ppng", ".ptif", ".ptiff", ".pbmp", ".pgif", ".pjpe", ".pjfif", ".pjt"};
+
+bool in_array(const string &value, const vector<string> &array) {
+    return std::find(array.begin(), array.end(), value) != array.end();
+}
+}  // namespace
+
+namespace mip {
+namespace file {
+
+FileFormatFactory::FileFormatFactory()
+{
+}
+
+std::shared_ptr<IFileFormat> FileFormatFactory::Create(std::shared_ptr<IStream> fileStream, const string& extension) {
+
+  if (extension.empty())
+  {
+    // TODO: add log
+    throw rmscore::exceptions::RMSInvalidArgumentException("The extention is empty");
```

Comment:
why do you throw RMSInvalidArgumentException? its not related to rms. i think there is built in exception for invalid argument